docs/src/index.js

```diff
@@ -9,6 +9,8 @@ import ReactPerf from 'react-addons-perf';
 import { render } from 'react-dom';
 import App from 'docs/src/components/App';
 
+require('core-js');
```

Comment:
@oliviertassinari I looked further into why I added `core-js`. This is [the line](https://github.com/callemall/material-ui/blob/next/docs/src/pages/component-demos/lists/CheckboxList.js#L46) that's breaking IE 11 in development mode:

```jsx
{Array.from({ length: 4 }, (v, k) => k).map(index =>
```


This is what the site looks like on IE 11 *without* `core-js` at `http://0.0.0.0:3000/component-demos/lists`:

<img width="1680" alt="screen shot 2017-06-12 at 2 56 07 am" src="https://user-images.githubusercontent.com/1571918/27028766-b0c2feea-4f1a-11e7-8bcc-deb9acd0091d.png">

Additionally, note that the site still works in production mode, so it won't create any issues of reproducibility adding it. Here's the same page on the hosted firebase instance for the `next` branch in IE 11:

<img width="1680" alt="screen shot 2017-06-12 at 3 04 06 am" src="https://user-images.githubusercontent.com/1571918/27029108-d102d5a8-4f1b-11e7-973d-91658879f5f9.png">


The trouble is, the markdown files *don't* get polyfilled in development mode. So if you don't want core-js, the documentation needs to be updated to use an API that will work in IE.

...But on the other hand, it wouldn't be a bad idea to be polyfilling in development mode in order to allow using API's like `Array.from`.

Let me know how you'd like to proceed. The documentation files can be updated.